Expose metrics port

brenner-tobias opened the pull request:
# Proposed Changes

Expose Metrics Server Port and add WebUI URL in Configuration

## Related Issues

#725


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced a metrics web interface for enhanced monitoring.
  - Updated port mapping for improved access to the metrics interface.

- **Configuration Improvements**
  - Reorganized development container settings for better clarity and structure.
  - Updated port configuration for development container to streamline access.
  - Added new network interface entries for metrics in multiple languages.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

elcajon commented on cloudflared/config.yaml:
```diff
@@ -18,6 +19,10 @@ map:
 options:
   external_hostname: ""
   additional_hosts: []
+ports:
+  36500/tcp: 36500
+ports_description:
+  36500/tcp: Metrics Web interface
```
I think we should add the description as [translation](https://developers.home-assistant.io/docs/add-ons/configuration#port-description-translations) to be consistent with the rest of our configuration descriptions.

I would suggest to set the port mapping to "null" to disable port forwarding per default.

elcajon reviewed (approved):
LGTM 👍🏼

brenner-tobias commented:
There is one issue with the way I implemented it: Since I added the `webui: http://[HOST]:[PORT:36500]/metrics` config, the link for the webui is visible in the add-on page. If you do not actively add the port though, clicking on the link loads a blank website `about:blank#blocked`

We do have three options:

1. Leave as it is right now
2. Remove the webui config, so the link is not shown and it is everyones own responsibility to configure the port and open the metrics link
3. Set the port to be present by default, so there are no issues for most users who do not disable it

I am leaning towards option 2, since this is a niche use case anyways, but I could also live with 3. 

@elcajon what do you think?

elcajon commented:
I would also tend towards option two, as otherwise the question would arise as to whether you need to forward the port on your router in order to be able to use the add-on and the tunnel at all...

felipecrs commented on cloudflared/config.yaml:
```diff
@@ -7,6 +7,7 @@ description: "Use a Cloudflare Tunnel to remotely
 url: "https://github.com/brenner-tobias/addon-cloudflared/"
 codenotary: dev@brenner.tech
 init: false
+webui: http://[HOST]:[PORT:36500]/metrics
```
![image](https://github.com/user-attachments/assets/c1ea1960-fab0-4184-8e8e-7d1d3c865385)

If your intention was to make a button show up here, I think it's missing ingress configuration too.

```yaml
ingress: true
ingress_port: 36500
ingress_stream: true # if websockets are needed, not sure
```

felipecrs replied:
Nevermind, just saw https://github.com/brenner-tobias/addon-cloudflared/pull/730.

And I agree, I don't think the button should be added indeed for such interface.

BTW it worked fine:

![image](https://github.com/user-attachments/assets/56c65ee5-81f0-4e31-b83e-0e4c21efa04e)

brenner-tobias replied:
Thanks, I think we are good with the current implementation, let me know if you have some suggestions for improvements.
